### DISCUSSION TITLE
Build on Asus Tinkerboard

### DISCUSSION THREAD

**Discussion by rabbit0864** [BlitterStudio/amiberry#908] [Q&A] [answered]
Building with make PLATFORM=RK3288, no obvious build errors but trying to start after gives "Segmentation fault"

Any ideas?

Many Thanks

**Comment by midwan** [comment, marked as answer]
It's been a while since I run it on a Tinkerboard myself, but it should still work.
On which distro are you testing this?

**Reply by rabbit0864** [reply]
Thanks for the getting back to me.

I've been using Armbian Buster Legacy 4.4.2 as the Focal versions don't have the hardware acceleration working properly - although I will attempt a build on Focal later.

The latest version of TinkerOS seems to be too out of date in terms of the versions of all the dependencies, as well as GCC etc.

I've also tried building v3.3 with the same result on Armbian. RetroPie also fails to build the Amiberry package.

Oddly, running via valgrind does allow the build to start up - albeit very slowly.

**Reply by midwan** [reply]
OK, I'll flash a fresh image of Armbian and try it out later today, see if I can recreate this. Did you install SDL2 from the repos or compile it yourself? That may be important.

**Reply by rabbit0864** [reply]
I installed from the repos - I also tried an older version of SDL2 - don't have the info to hand as to the versions at the moment though

**Reply by midwan** [reply]
Did a quick test on Armbian buster, and it seems to work for me. I'll do a test on the latest Focal as well, a little later.

**Reply by rabbit0864** [reply]
Wonder what I'm doing wrong then?

Builds OK, but refuses to run.

**Reply by midwan** [reply]
Seems to work normally under Focal desktop also. Are you running it under X/Desktop? I just used the latest Armbian desktop image, installed the dependencies and compiled it. Nothing special.

**Reply by rabbit0864** [reply]
Running in XFCE desktop.

First ran 

sudo apt update && sudo apt upgrade

I followed instructions to install dependencies then

git clone https://github.com/midwan/amiberry
cd amiberry
make PLATFORM=RK3288

Build seems to work OK then I run 
./amiberry
and get an immediate Segmentation Fault

**Reply by rabbit0864** [reply]
Which buster link did you use / download?

**Reply by midwan** [reply]
The buster image was the previous one I had on that SD card, which I tested first. Then I downloaded the latest Focal Desktop from here: https://www.armbian.com/tinkerboard/ and flashed it. I only set up the defaults (user, password, wifi, locale), installed the dependencies as documented on the Readme, cloned Amiberry and compiled it. Then I run it from there.

**Reply by rabbit0864** [reply]
OK, got it working in Focal, thank you. Still no idea whats going wrong in Buster - but the OS is much snappier because the drivers etc are not up to scratch in Focal.

**Reply by rabbit0864** [reply]
So, works in Focal and the Buster Current builds of Armbian.

I was attempting to build in the Buster Legacy branch - the reason being that there is no hardware acceleration for the GPU in the Focal or current builds - so although it is working, games are completely unplayable as the refresh rate is too slow. Running in a window is quicker.

So - what do I need to do to build Amiberry in the legacy build of Armbian - or will it never work?

**Reply by midwan** [reply]
I'm afraid that's up to the distro maintainers to deal with, Amiberry will just use the SDL2 framework with whatever back-end is available. Some platforms have better driver support than others, but there's not much I can do about that from my side... :(
